src/item.cpp

```diff
@@ -5778,22 +5778,68 @@ bool item::process_food( player * /*carrier*/, const tripoint &pos )
     if( item_tags.count( "FROZEN" ) > 0 && item_counter > 500 && type->comestible->parasites > 0 ) {
         item_tags.insert( "NO_PARASITES" );
     }
+
+    // minimum is 0 - takes into account that process() takes --1 counter per turn regardless
+    // div by 5 means every 5 degrees of difference equals 1 point of ratio
     /* cache g->get_temperature( item location ). It is used a minimum of 3 times, no reason to recalculate. */
-    const auto item_local_temp = g->get_temperature( pos );
-    unsigned int diff_freeze = abs( item_local_temp - FREEZING_TEMPERATURE );
-    diff_freeze = diff_freeze < 1 ? 1 : diff_freeze;
-    diff_freeze = diff_freeze > 10 ? 10 : diff_freeze;
-
-    unsigned int diff_cold = abs( item_local_temp - FRIDGE_TEMPERATURE );
-    diff_cold = diff_cold < 1 ? 1 : diff_cold;
-    diff_cold = diff_cold > 10 ? 10 : diff_cold;
-    // environment temperature applies COLD/FROZEN flags to food
-    if( item_local_temp <= FRIDGE_TEMPERATURE ) {
-        g->m.apply_in_fridge( *this, item_local_temp );
-    } else if ( item_tags.count( "FROZEN" ) > 0 && item_counter > diff_freeze ) {
-        item_counter -= diff_freeze;
-    } else if( item_tags.count( "COLD" ) > 0 && item_counter > diff_cold ) {
-        item_counter -= diff_cold;
+    const auto temp = g->get_temperature( pos );
+    unsigned int diff_freeze = abs( temp - FREEZING_TEMPERATURE ) / 5 ;
+    diff_freeze = clamp( diff_freeze, static_cast<unsigned int>(0) , static_cast<unsigned int>(3) ); //effective 1-4
+
+    // div by 5 means every 5 degrees of difference equals 1 point of ratio
+    unsigned int diff_cold = abs( temp - FRIDGE_TEMPERATURE ) / 5;
+    diff_cold = clamp( diff_cold, static_cast<unsigned int>(0), static_cast<unsigned int>(3) ); //effective 1-4
+
+    // exclusions for items being actively cooled to be affected by outside temperatures
+    //const optional_vpart_position vp = g->m.veh_at( pos );
+    const optional_vpart_position vp = g->m.veh_at( pos );
+    bool in_active_cooler = false;
+    bool fridge_here = false;
+    bool freezer_here = false;
+    if( vp ) {
+        fridge_here = vp->vehicle().has_part( pos, "FRIDGE", true );
+        freezer_here = vp->vehicle().has_part( pos, "FREEZER", true );
+        if(fridge_here) { add_msg( "Fridge" ); } else {add_msg( "NOT Fridge" );}
+        if(freezer_here) { add_msg( "Freezer" ); } else {add_msg( "NOT Freezer" );}
+    }
+
+    if( vp && ( fridge_here || freezer_here ) ) {
+        std::vector<vehicle_part *> parts = vp->vehicle().get_parts( pos, "FRIDGE", true );
+        if( parts.empty() ) { 
+            parts = vp->vehicle().get_parts( pos, "FREEZER", true );
+        }
```

Comment:
You've already checked if `vp` is null L@5799, why not just put this inside that block?

Then for `parts` you've already checked if veh has a FRIDGE or FREEZER at L@5800, why not use those bools to figure out what `parts` should be instead of these shenanigans?  You also don't need the `part` var at L@5812 because you're just grabbing the items off it.